### PR TITLE
Added buttons to configuration pages

### PR DESCRIPTION
**Pull request by luigifab**
### Description

A new simple feature for _System > Configuration_ pages.

For example, add the following `<buttons>` tag in _app/code/core/Mage/Core/etc/system.xml_ line ~731:
```xml
        <general translate="label" module="core">
            <label>General</label>
            [...]
            <buttons>
                <products translate="title">
                    <title>Products</title>
                    <action>adminhtml/catalog_product/index</action>
                </products>
            </buttons>
```
Tags: `<title>` + `<action>` or `<onclick>` + `<class>` + `<sort_order>`.

image

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Adminhtml/Block/System/Config/Edit.php [hunk 1]
```diff
@@ -65,7 +65,22 @@ protected function _prepareLayout()
      */
     public function getSaveButtonHtml()
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Edit.php:66]: I think it's better to create a separate method, something like `public function getAdditionalButtons()`, and call it in `app\design\adminhtml\default\default\template\system\config\edit.phtml`.

Comment (luigifab) [reply app/code/core/Mage/Adminhtml/Block/System/Config/Edit.php]: done

**Review by kiatng — APPROVED** [review]
Tested, it works.

### A note on adding buttons in system.xml with this PR
You need to add the configuration element `<buttons>` as a sibling to `<groups>` in your module. As an example, for a module called 'tpa':
```xml
<config>
    <sections>
        <tpa translate="label comment" module="tpa">
            <tab>some_tab</tab>
            <label>TPA Module</label>
            <sort_order>30</sort_order>
            <show_in_default>1</show_in_default>
            <show_in_website>0</show_in_website>
            <show_in_store>0</show_in_store>
            <groups>
                  <!-- blah blah. -->
            </groups>
           <!-- Begin : add buttons -->
            <buttons>
                <products translate="title">
                    <title>Products</title>
                    <action>adminhtml/catalog_product/index</action>
                </products>
            </buttons>
           <!-- End : add buttons -->
        </tpa>
    </sections>
</config>
```

The buttons will be rendered only when you navigate to your module in the system configuration. It will not rendered when you navigate away from it.

**Comment by fballiano** [comment]
should we have some documentation somewhere about this? I'm actually wondering where :-\

**Comment by luigifab** [comment]
Perhaps I can simply add buttons in _System > Configuration > Catalog > Catalog_ to Products and Categories, and in _System > Configuration > Customer > Customer_ to Customers?

**Comment by luigifab** [comment]
I'm thinking that `<sort_order>` for buttons is missing.
And perhaps `<acl>`...

**Comment by ADDISON74** [comment]
What is the role of these buttons and how do they actually help us? As far as I can see, there are links to the "Manage Product" and "Manage Categories" sections. Why wouldn't I access them as before from the menu? I don't like to see them in the configuration page that has nothing to do with it.

**Comment by fballiano** [comment]
Because of @ADDISON74 latest comment (I agree 100%) and the ACL problem discussed https://github.com/OpenMage/magento-lts/pull/2678 I'll close this one.

feel free to reopen in case something changed

**Pull request closed without merging**